manager/controllers/app/plotter_generator.go

```diff
@@ -181,18 +184,32 @@ func (p *PlotterGenerator) addInMemoryStep(element *ResolvedEdge, datasetID stri
 	if lastStepAPI == nil {
 		assetID = datasetID
 	}
-	steps = append(steps, app.DataFlowStep{
-		Cluster:  element.Cluster,
-		Template: string(element.Module.Spec.Capabilities[element.CapabilityIndex].Capability),
-		Parameters: &app.StepParameters{
-			Source: &app.StepSource{
-				AssetID: assetID,
-				API:     lastStepAPI,
+	if flowType == taxonomy.WriteFlow {
```

Comment:
The hope was to not need flow specific if statements in the code, so that in the future we can support new flows without making changes